test/end2end/base_environment.py

```diff
@@ -299,3 +421,97 @@ def get_all_tablet_types(self, keyspace, num_shards):
     for shard_name in sharding_utils.get_shard_names(num_shards):
       tablet_info += self.get_tablet_types_for_shard(keyspace, shard_name)
     return tablet_info
+
+  def backup(self, tablet_name):
+    """Wait until all tablets report healthy status.
+
+    Args:
+      tablet_name: Name of tablet to backup. (string)
+
+    Raises:
+      VitessEnvironmentError: Raised if unsupported
+    """
+    raise VitessEnvironmentError(
+        'Backup unsupported in this environment')
+
+  def drain_tablet(self, tablet_name, duration_s=600):
+    """Add a drain from a tablet.
+
+    Args:
+      tablet_name: vttablet to drain (string).
+      duration_s: how long to have the drain exist for, in seconds (int).
+
+    Raises:
+      VitessEnvironmentError: Raised if unsupported
+    """
+    raise VitessEnvironmentError(
+        'drain unsupported in this environment')
+
+  def is_tablet_drained(self, tablet_name):
+    """Checks whether a tablet is drained.
+
+    Args:
+      tablet_name: vttablet to drain (string).
+
+    Raises:
+      VitessEnvironmentError: Raised if unsupported
+    """
+    raise VitessEnvironmentError(
+        'is tablet drained unsupported in this environment')
```

Comment:
I think you should use underscores to refer to method names. These exceptions sound like you don't quite understand English.